jwst/pipeline/calwebb_tso3.py

```diff
@@ -58,10 +58,15 @@ def process(self, input):
         self.asn_id = input_models.meta.asn_table.asn_id
 
         input_exptype = None
+        input_tsovisit = None
         # Input may consist of multiple exposures, so loop over each of them
         for cube in input_models:
             if input_exptype is None:
                 input_exptype = cube.meta.exposure.type
+
+            if input_tsovisit is None:
+                input_tsovisit = cube.meta.visit.tsovisit
```

Comment:
Will every cube in `input_models` have the same exposure type and tsovisit flag?

Reply:
I asked Jonathan about that very question. An association containing TSO data will only have tso  data and will have the same o exp_type and tsovisit.